service/src/main/kotlin/no/nav/su/se/bakover/service/kontrollsamtale/KontrollsamtaleService.kt

```diff
@@ -0,0 +1,111 @@
+package no.nav.su.se.bakover.service.kontrollsamtale
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.client.person.MicrosoftGraphApiOppslag
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.log
+import no.nav.su.se.bakover.common.persistence.SessionFactory
+import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.Person
+import no.nav.su.se.bakover.domain.brev.LagBrevRequest
+import no.nav.su.se.bakover.domain.dokument.Dokument
+import no.nav.su.se.bakover.domain.oppgave.OppgaveConfig
+import no.nav.su.se.bakover.service.brev.BrevService
+import no.nav.su.se.bakover.service.brev.KunneIkkeLageBrev
+import no.nav.su.se.bakover.service.oppgave.OppgaveService
+import no.nav.su.se.bakover.service.person.PersonService
+import no.nav.su.se.bakover.service.sak.SakService
+import java.time.Clock
+import java.time.LocalDate
+import java.util.UUID
+
+interface KontrollsamtaleService {
+    fun kallInn(sakId: UUID, saksbehandler: NavIdentBruker): Either<KunneIkkeKalleInnTilKontrollsamtale, Unit>
+}
+
+class KontrollsamtaleServiceImpl(
+    private val sakService: SakService,
+    private val personService: PersonService,
+    private val brevService: BrevService,
+    private val oppgaveService: OppgaveService,
+    private val microsoftGraphApiOppslag: MicrosoftGraphApiOppslag,
+    private val sessionFactory: SessionFactory,
+    private val clock: Clock,
+) : KontrollsamtaleService {
+
+    override fun kallInn(
+        sakId: UUID,
+        saksbehandler: NavIdentBruker,
+    ): Either<KunneIkkeKalleInnTilKontrollsamtale, Unit> {
+        val sak = sakService.hentSak(sakId).getOrElse {
+            log.error("Fant ikke sak for sakId $sakId")
+            return KunneIkkeKalleInnTilKontrollsamtale.FantIkkeSak.left()
+        }
+
+        val person = personService.hentPerson(sak.fnr).getOrElse {
+            log.error("Fant ikke person for fnr: ${sak.fnr}")
+            return KunneIkkeKalleInnTilKontrollsamtale.FantIkkePerson.left()
+        }
+
+        val saksbehandlerNavn = microsoftGraphApiOppslag.hentNavnForNavIdent(saksbehandler).getOrElse {
+            log.error("Fant ikke saksbehandlernavn for saksbehandler: $saksbehandler")
+            return KunneIkkeKalleInnTilKontrollsamtale.KunneIkkeHenteNavnForSaksbehandlerEllerAttestant.left()
+        }
+
+        val dokument = lagDokument(person, saksbehandlerNavn, sakId).getOrElse {
+            log.error("Klarte ikke lage dokument for innkalling til kontrollsamtale på sakId $sakId")
+            return KunneIkkeKalleInnTilKontrollsamtale.KunneIkkeGenerereDokument.left()
+        }
+
+        return Either.catch {
+            sessionFactory.withTransactionContext {
+                brevService.lagreDokument(dokument, it)
```

Comment:
Rev generell kommentar; trenger vi noe som syr dette sammen? Eller er det greit nok at disse bare "flyter fritt" (inntil videre)?

Reply:
Tenker vi kan la det gå en iterasjon eller to uten at vi syr det spesielt sammen. Dette er en erstatning for brev de p.t. sender manuelt fra Gosys, så vi gjør det bedre for dem, uten at det blir perfekt (men der kan vi nok avvente litt og se på hvordan resten av domenet for kontrollsamtale bør se ut, ref. kommentaren over).